src/jit/codegenxarch.cpp

```diff
@@ -8219,28 +8251,38 @@ CodeGen::genPutStructArgStk(GenTreePtr treeNode, unsigned baseVarNum)
                     }
                 }
                 break;
+
+            case TYPE_GC_REF:    // Is an object ref
+            case TYPE_GC_BYREF:  // Is an interior pointer - promote it but don't scan it
+                {
+                    // We have a GC (byref or ref) pointer
+                    // TODO-Amd64-Unix: Here a better solution (for code size and CQ) would be to use movsq instruction,
+                    // but the logic for emitting a GC info record is not available (it is internal for the emitter only.)
+                    // See emitGCVarLiveUpd function. If we could call it separately, we could do instGen(INS_movsq); and emission of gc info.
+
+                    getEmitter()->emitIns_R_AR(ins_Load(srcAddr->gtGetOp1()->TypeGet()), emitTypeSize(srcAddr->gtGetOp1()), REG_RCX, REG_RSI, 0);
+                    getEmitter()->emitIns_S_R(ins_Store(srcAddr->gtGetOp1()->TypeGet()),
+                                                        emitTypeSize(srcAddr->gtGetOp1()),
+                                                        REG_RCX,
+                                                        baseVarNum,
+                                                        ((copiedSlots + putArgStk->gtSlotNum) * TARGET_POINTER_SIZE));
+
+                    getEmitter()->emitIns_R_I(INS_add, emitTypeSize(srcAddr->gtGetOp1()), REG_RSI, TARGET_POINTER_SIZE);
+                    getEmitter()->emitIns_R_I(INS_add, EA_8BYTE, REG_RDI, TARGET_POINTER_SIZE);
+                    copiedSlots++;
+                    gcPtrCount--;
```

Comment:
the purpose of gcPtrCount local is to assert at the end of copying of struct that it is zero.  Please add an assert  after the while-loop that gcPtrCount == 0.